src/plugins/intel_cpu/src/weights_cache.cpp

```diff
@@ -39,7 +39,11 @@ WeightsSharing::SharedMemory::Ptr WeightsSharing::findOrCreate(const std::string
         std::unique_lock<std::mutex> lock(guard);
         auto found = sharedWeights.find(key);
 
-        if (found == sharedWeights.end() || !((ptr = found->second) && (newPtr = ptr->sharedMemory.lock()))) {
+        auto isInvalid = [&]() {
+            return found == sharedWeights.end() || !((ptr = found->second) && (newPtr = ptr->sharedMemory.lock()));
```

Comment:
Could you please split the logic the same way you have done it below

Reply:
Done